testcases/kernel/syscalls/recvmmsg/recvmmsg01.c

```diff
@@ -0,0 +1,2250 @@
+	}
+
+	n_selected *= N_REPEAT;
+	tst_res(TINFO, "Running %d non-error SOCK_SEQPACKET "
+		"tests sharing a connection (N_REPEAT = %d):\n",
+		n_selected, N_REPEAT);
+	con_init_seqpacket(&con);
+	for (i = 0; i < n_selected; ++i) {
+		callp = call_tests_selected[i];
+		error = call_go(callp, &con);
+		test_assert(!error);
+	}
+	con_deinit(&con);
+
+	/*
+	 * Now run them with recvmmsg(2)
+	 */
+
+	int sendmulti;
+	for (sendmulti = 0; sendmulti <= 1; ++sendmulti) {
+		tst_res(TINFO, "Running %d non-error SOCK_SEQPACKET "
+			"tests with a single recvmmsg(2) call "
+			"(N_REPEAT = %d) and %s:\n",
+			n_selected, N_REPEAT,
+			sendmulti ? "a single sendmmsg(2) call"
+				  : "multiple sendmsg(2) calls");
+		con_init_seqpacket(&con);
+		con_make_vectored(&con, sendmulti > 0, n_selected,
+				  call_tests_selected, smc_vec, rmc_vec);
+		callp = call_tests_selected[0];
+		error = call_go(callp, &con);
+		test_assert(!error);					
+		con_deinit(&con);
+	}
+
+	/*
+	 * Repeat the first test iov_max with recvmmsg(2)
+	 */
+
+	n_selected = iov_max;
+	callp = call_tests_selected[0];
+	for (i = 1; i < n_selected; ++i)
+		call_tests_selected[i] = callp;
+
+	for (sendmulti = 0; sendmulti <= 1; ++sendmulti) {
+		tst_res(TINFO, "Running %d non-error SOCK_SEQPACKET "
+			"tests with a single recvmmsg(2) call "
+			"and %s:\n", n_selected,
+			sendmulti ? "a single sendmmsg(2) call"
+				  : "multiple sendmsg(2) calls");
+		con_init_seqpacket(&con);
+		con_make_vectored(&con, sendmulti > 0, n_selected,
+				  call_tests_selected, smc_vec, rmc_vec);
+		callp = call_tests_selected[0];
+		error = call_go(callp, &con);
+		test_assert(!error);					
+		con_deinit(&con);
+	}
+}
+
+/*
+ *	The LTP coding convention is for main() not to be provided by
+ *	the test programs, for easier debugging you might want to use
+ *	this main() function (for easier testing instead of having to go
+ *	find the LTP library code that the LTP provided main() function
+ *	uses).
+ *
+ *	If you want to use this main() function add this #define:
+ *		#define TST_NO_DEFAULT_MAIN
+ *	Prior to this #include:
+ *		#include "tst_test.h"
+ *	at the start of this file.
+ *
+ *	Another reason you might want to do this is to pass one or more:
+ *		-v
+ *	to the test program to increse its verbosity level, alternatively
+ *	you can initialize test_verbose to a non-zero value and recompile.
+ */
+
+#ifdef TST_NO_DEFAULT_MAIN
+
+static void tst_parse_opt(int argc, char *argv[]);
+
+int main(int argc, char *argv[])
+{
+	tst_parse_opt(argc, argv);
+	tests_setup();
+	tests_run();
+	exit(ensure_failures > 0 ? 1 : 0);
+}
+
+static void tst_parse_opt(int argc, char *argv[])
+{
+	prog_name = argv[0];
+	char *lasts = strrchr(prog_name, '/');
+	if (lasts)
+		prog_name = lasts + 1;
+	TCID = prog_name;
+	while (--argc >= 1) {
+		char *arg = *++argv;
+		if (strcmp(arg, "-v") == 0)
+			++test_verbose;
+		else {
+			fprintf(stderr, "usage: %s [-v]\n", prog_name);
+			exit(1);
+		}
+	}
+}
+
```

Comment:
You can pass options to LTP testcases as well, all you need to do is to fill in a structure and pass a pointer to it in the tst_test structure:

https://github.com/linux-test-project/ltp/wiki/Test-Writing-Guidelines#225-test-specific-command-line-options

So NACK to this hackery.